### PR TITLE
Work on automated tests for the 'io' library

### PR DESCRIPTION
**Pull request by pedro-w**
This fixes Testworks fails for the io library. In some cases the test was changed, in some the library.
Note that there are still a lot of tests not implemented.

### CODE REVIEW

File: sources/io/streams/sequence-stream.dylan [hunk 1]
```diff
@@ -95,7 +95,8 @@ define method initialize
     stream.final-position := _start
   else
     stream.final-position := size(stream-sequence(stream))
-  end
+  end;
+  stream.stream-element-type := element-type(stream-sequence(stream));
 end method initialize;
```
Comment (cgay) [inline sources/io/streams/sequence-stream.dylan]: I'm not sure about this one. This seems to override the value given to the element-type: init-keyword. I found this in the docs:

"The element-type init-keyword specifies the type of the elements in the file named by filename. This allows file elements to be represented abstractly; for instance, contiguous elements could be treated as a single database record."

So maybe the intention was for the elements of the underlying sequence to be converted to the type given by the element-type: init-keyword?

File: sources/common-dylan/format.dylan [hunk 2]
```diff
@@ -713,7 +713,6 @@ define function print-method
   print-string(buffer, object-class-name(object));
   print-string(buffer, ": ");
   print-string(buffer, primitive-name(object));
-  print-string(buffer, " ");
```
[the remark this answers is not in the document]
Comment (pedro-w) [reply sources/common-dylan/format.dylan]: Agreed. In `print-signature` (actually `print-signature-internal`) it starts by printing " (" so we were getting two spaces here.
https://github.com/dylan-lang/opendylan/blob/f8e2579fe0a614c6dfebe1984ac94d2464b4bfba/sources/common-dylan/format.dylan#L754-L760

File: sources/io/streams/sequence-stream.dylan (hunk 1 above)
Comment (pedro-w) [reply sources/io/streams/sequence-stream.dylan]: Interesting point. I assume [this](https://opendylan.org/documentation/library-reference/io/streams.html#options-when-creating-file-streams) is your reference?
It's odd because it talks about treating contiguous elements as a database record but goes on to list the possible element types as `<byte>`, `<byte-character>` and `<unicode-character>`.
Also, that reference is specific to file streams; there doesn't seem to be an element-type: keyword for sequence streams (https://opendylan.org/documentation/library-reference/io/streams.html#io:streams:[sequence-stream])
This makes sense because a file on disk is just an array of bytes whereas a sequence must have an element type (even if it's `<object>`), so the user would have to impose a type/format onto the former but not the latter.
What do you think?

Comment (cgay) [reply sources/io/streams/sequence-stream.dylan]: I failed to read the list of element types somehow. That makes me think that the bit about database records is just wrong (or at best misleading) and should be deleted.  Even if it was the original intention, I don't think functionality like that has any place in the streams code. Interpretation of the data should probably be handled at a higher level.  (Go doesn't seem to even do the byte -> text conversion for you; you can call string([]byte) if you want a string.)

I think when I was looking at this I saw that element-type: is specified by `<basic-stream>`, meaning that technically you could use it when creating a `<sequence-stream>`, but it probably would have no effect.

In other words, I think this change is fine. Perhaps remove that database verbiage from the docs while you're at it?

File: sources/common-dylan/format.dylan (hunk 2 above)
[the remark this answers is not in the document]
Comment (cgay) [reply sources/common-dylan/format.dylan]: I see, thanks.

File: sources/io/streams/sequence-stream.dylan [hunk 3]
```diff
@@ -188,16 +188,23 @@ define method read
...
-    n := src-n
+  else
+    let elements = copy-sequence(seq, start: pos, end: pos + n);
+    stream.current-position := pos + n;
+    elements
   end;
-  let elements = copy-sequence(seq, start: pos, end: pos + n);
-  stream.current-position := pos + n;
-  elements
 end method read;
```
[the remark this answers is not in the document]
Comment (cgay) [reply sources/io/streams/sequence-stream.dylan]: heh

[the remark this answers is not in the document]
Comment (pedro-w) [reply sources/io/streams/sequence-stream.dylan]: There aren't many tests for this part of the library, and when I added them some more problems became apparent. I will do more on this.
